src/forum/forum_post.rs

```diff
@@ -167,33 +169,52 @@ pub struct PostTreeNode {
 }
 
 impl PostTreeNode {
-    pub fn build_tree(conn: &rusqlite::Connection, parent_id: usize) -> ForumResult<Vec<Self>> {
+    pub fn build_tree(conn: &rusqlite::Connection, post_id: usize) -> ForumResult<Vec<Self>> {
+        let post = ForumPost::get(conn, post_id).unwrap();
+        let root_id = post.root_id.unwrap_or(post_id); // due to how OP's are saved, the OP.root_id != OP.id
         let mut stmt = conn
             .prepare(
                 "
             SELECT id, root_id, parent_id, created_at, deleted_at, author, message
             FROM forum_posts
-            WHERE parent_id = ?1
+            WHERE root_id = ?1
             ORDER BY created_at ASC
             ",
             )
             .map_err(ForumError::DatabaseError)?;
 
-        let mut nodes = stmt
-            .query_map([parent_id], |row| {
-                Ok(PostTreeNode {
-                    post: ForumPost::get_from_db(row).unwrap(),
-                    replies: Vec::new(),
-                })
-            })
+        let posts = stmt
+            .query_map([root_id], |row| Ok(ForumPost::get_from_db(row).unwrap()))
```

Comment:
Don't use `.unwrap()` when the function returns a `Result<...>`, use the `?` operator.